src/errors.rs

```diff
@@ -268,222 +275,344 @@ impl Reportable for DocumentError {
+            doc_line,
+            rx_snippet,
+            doc_snippet,
         })
     }
 }
 
-impl Reportable for ContentError {
+impl Reportable for TypeMismatchError {
     fn short_msg(&self) -> String {
-        let error_type =
-            ShellText::ErrorColor(Box::new(ShellText::Literal("Content Error".to_string())));
-        let rx_info = ShellText::Underlined(Box::new(ShellText::Literal(format!(
-            "{} line {}",
-            self.rx_file, self.rx_line
-        ))));
-        let document_info = ShellText::Underlined(Box::new(ShellText::Literal(format!(
-            "{} line {}",
-            self.document_file, self.document_line
-        ))));
-        let mut message = String::new();
-
-        for pair in &self.match_pairs {
-            if !ContentMatchPair::is_match(&pair) {
-                message = match pair {
-                    &ContentMatchPair(PromptToken::Mandatory, None) => {
-                        ShellText::MessageColor(Box::new(ShellText::Literal(
-                            "No match found for Mandatory Prompt".to_string(),
-                        ))).to_string()
-                    }
-                    &ContentMatchPair(PromptToken::Literal(ref required), None) => {
-                        let required_content = ShellText::ErrorColor(Box::new(ShellText::Literal(
-                            required.clone(),
-                        ))).to_string();
-                        format!(
-                            "The required literal content '{}' was missing.",
-                            required_content
-                        )
-                    }
-                    &ContentMatchPair(PromptToken::Literal(ref required), Some(ref found)) => {
-                        let required_content = ShellText::ErrorColor(Box::new(ShellText::Literal(
-                            required.clone(),
-                        ))).to_string();
-                        let found_content = ShellText::ErrorColor(Box::new(ShellText::Literal(
-                            found.clone(),
-                        ))).to_string();
-                        format!(
-                            "The required literal content '{}' does not match '{}'",
-                            required_content, found_content
-                        )
-                    }
-                    &ContentMatchPair(PromptToken::None, Some(ref content)) => {
-                        let found_content = ShellText::ErrorColor(Box::new(ShellText::Literal(
-                            content.clone(),
-                        ))).to_string();
-                        format!("Superfluous content found: '{}'", found_content)
-                    }
-                    _ => "Content Mismatch".to_string(),
-                };
-                break;
-            }
-        }
+        let error_type = format!(
+            "{}Type Mismatch Error{}",
+            color::Fg(color::Red),
+            color::Fg(color::Reset)
+        );
```

Comment:
This formatting pattern is repeated a lot, good opportunity for a little helper.

Reply:
helpers added